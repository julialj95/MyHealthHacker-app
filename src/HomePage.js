import React from "react";

class HomePage extends React.Component {
  render() {
    return (
      <div>
        <h1>Home Page Here</h1>
      </div>
    );
  }
}

export default HomePage;
