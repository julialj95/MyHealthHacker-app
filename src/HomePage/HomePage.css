.about {
  text-align: center;
  margin-left: 10%;
  margin-right: 10%;
}

.home_header {
  margin-top: 0px;
  font-size: 24px;
}

.home_title {
  font-size: 18px;
}
.home_container {
  display: flex;
  flex-direction: column;
}
.background_wrapper {
  background-color: #1a0135;
  padding: 0;
  margin: 0;
  width: 100%;
  display: flex;
  border-radius: 15px;
  margin: 20px 0 20px 0;
  display: flex;
}
.img_text_block {
  display: flex;
  flex-direction: column;
  flex: 1;
  margin: 5px;
}

.home_p {
  font-size: 14px;
  flex-grow: 1;
  color: white;
  padding: 15px;
  border-radius: 5px;
  margin-block-start: 0;
  margin-block-end: 0;
  margin-inline-start: 0;
  margin-inline-end: 0;
}
.home_img {
  flex-grow: 1;
  border: solid 3px #1a0135;
  max-height: 200px;
  max-width: 200px;

  flex-basis: auto;
  border-radius: 15px;
  margin: auto;
}

.reverse {
  flex-wrap: wrap-reverse;
}
.link {
  color: white;
}

@media all and (min-width: 350px) {
  .home_img {
    max-width: 250px;
    max-height: 250px;
  }
}
@media all and (min-width: 650px) {
  .img_text_block {
    flex-direction: row;
  }
  .home_header {
    font-size: 28px;
  }
  .home_title {
    font-size: 24px;
  }

  .home_p {
    font-size: 18px;
  }
  .home_img {
    margin: auto;
    max-width: 275px;
    max-height: 275px;
  }

  .reverse {
    flex-direction: row-reverse;
  }
}

@media all and (min-width: 768px) {
  .home_header {
    font-size: 32px;
  }
  .home_title {
    font-size: 28px;
  }

  p,
  .home_p {
    font-size: 24px;
  }
}

@media all and (min-width: 960px) {
  p,
  .home_p {
    font-size: 28px;
  }
  .home_img {
    max-width: 325px;
    max-height: 325px;
  }
}
