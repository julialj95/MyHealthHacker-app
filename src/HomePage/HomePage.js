import React from "react";
import "./HomePage.css";

class HomePage extends React.Component {
  render() {
    return (
      <main className="about">
        <h1>MyHealthHacker</h1>
        <section>
          <h2>Why Track Your Health Habits?</h2>
          <p>
            Studies show that becoming aware of habits is the first step towards
            changing them. MyHealthHacker will help you to become aware of your
            health habits and track how different behaviors make you feel
            mentally and physically!{" "}
          </p>

          <p>
            Each day, you can track your sleep patterns, your physical activity,
            and mental health. Over time, you will begin to notice patterns and
            correlations between what you did and how you felt. This will allow
            you to adjust your habits to help you feel your best!
          </p>

          <h2>How To Use MyHealthHacker</h2>
          <p>
            Create an account and get ready to submit your first log. You can
            log in and check your logs any time!
          </p>
        </section>
      </main>
    );
  }
}

export default HomePage;
