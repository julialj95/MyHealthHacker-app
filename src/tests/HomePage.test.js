import React from "react";
import ReactDOM from "react-dom";
import HomePage from "../HomePage/HomePage";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <HomePage />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Router>
        <HomePage />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
