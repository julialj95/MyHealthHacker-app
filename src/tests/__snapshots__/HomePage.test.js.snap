// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  className="about"
>
  <h1
    className="home_header"
  >
    MyHealthHacker
  </h1>
  <section
    className="home_container"
  >
    <h2
      className="home_title"
    >
      Why Track Your Health Habits?
    </h2>
    <div
      className="background_wrapper"
    >
      <div
        className="img_text_block"
      >
        <p
          className="home_p"
        >
          The first step towards improving your habits is to become aware of them. MyHealthHacker will help you to increase your awareness of your health habits and track how different behaviors make you feel mentally and physically!
           
        </p>
        <img
          alt="wellness"
          className="home_img"
          src="wellness_graphic.png"
        />
      </div>
    </div>
    <h2
      className="home_title"
    >
      What You Will Notice
    </h2>
    <div
      className="background_wrapper"
    >
      <div
        className="img_text_block"
      >
        <p
          className="home_p"
        >
          Each day, you can track your sleep patterns, your physical activity, and mental health. Over time, you will begin to notice patterns and correlations between what you did and how you felt. This will allow you to adjust your habits to help you feel your best!
        </p>
        <img
          alt="graph"
          className="home_img"
          src="bar_graph.png"
        />
      </div>
    </div>
    <h2
      className="home_title"
    >
      How To Use MyHealthHacker
    </h2>
    <div
      className="background_wrapper"
    >
      <p
        className="home_p"
      >
        <a
          className="link"
          href="/signup"
          onClick={[Function]}
        >
          Create an account
        </a>
         
        or
         
        <a
          className="link"
          href="/login"
          onClick={[Function]}
        >
          login
        </a>
         
        with a demo account to submit your first log. You can log in to check, submit and edit your logs any time!
      </p>
    </div>
  </section>
</main>
`;
